motoman_driver/include/motoman_driver/simple_message/motoman_read_single_io_reply.h

```diff
@@ -57,15 +57,19 @@ namespace io_ctrl_reply
 /**
  * \brief Enumeration of Read Single IO reply result codes.
  */
-namespace ReadSingleIOReplyResults
+namespace ReadSingleIOReplyResultCodes
 {
-enum ReadSingleIOReplyResult
+enum ReadSingleIOReplyResultCode
 {
-  FAILURE    = 0,
-  SUCCESS    = 1
+  SUCCESS               =    0,
+  READ_ADDRESS_INVALID  = 1001, // The ioAddress cannot be read on this controller
+  WRITE_ADDRESS_INVALID = 1002, // The ioAddress cannot be written to on this controller
+  WRITE_VALUE_INVALID   = 1003, // The value supplied is not a valid value for the addressed IO element
+  READ_API_ERROR        = 1004, // mpReadIO returned -1
+  WRITE_API_ERROR       = 1005, // mpWriteIO returned -1
 };
 }
-typedef ReadSingleIOReplyResults::ReadSingleIOReplyResult ReadSingleIOReplyResult;
+typedef ReadSingleIOReplyResultCodes::ReadSingleIOReplyResultCode ReadSingleIOReplyResultCode;

```

Comment:
Out of curiosity, why are the result code classes wrapped in a namespace if we are just going to `typedef` the namespace away?

Reply:
This is essentially cargo-cult. I'm just maintaining/duplicating what was there already.

Not sure why it was done this way, perhaps that was the best available at the time.